enhancements/proposals/2490-scylladbmonitoring-external-prometheus/README.md

```diff
@@ -0,0 +1,602 @@
+kind: ScyllaDBMonitoring
+metadata:
+  name: example
+spec:
+  type: Platform
+  endpointsSelector:
+    matchLabels:
+      app.kubernetes.io/name: scylla
+      scylla-operator.scylladb.com/scylla-service-type: member
+      scylla/cluster: replace-with-your-scyllacluster-name
+  components:
+    prometheus:
+      mode: UserManaged
+    grafana:
+      datasources:
+        - access: proxy
+          url: "https://external-prometheus-address:9090"
+          prometheus:
+            auth:
+              bearerTokenSecretName: prometheus-bearer-token
+            tls:
+              server:
+                caCertConfigMapName: prometheus-serving-ca
+                insecureSkipVerify: false
+              client:
+                secretName: prometheus-client-certs
+```
+
+### Validation
+
+#### Admission Webhook
+
+The Operator will validate in its admission webhook the `ScyllaDBMonitoring` objects to comply with the following rules:
+
+- If `spec.components.prometheus.mode` is set to `UserManaged`, `spec.components.grafana.datasources` must be set
+- If `spec.components.grafana.datasources` is set, it must contain exactly one datasource of type `Prometheus`.
+- Immutability of `spec.components.prometheus.mode` field.
+
+### Test Plan
+
+- Validation rules will be covered in unit tests.
+- Functions generating `ScyllaDBMonitoring` child resources will be covered in unit tests.
+- A regular E2E test will be added to verify that it's possible to use `ScyllaDBMonitoring` with `UserManaged`
+  Prometheus mode and configure the managed Grafana instance to use an external Prometheus instance as a data source
+  (with mTLS).
+- An OpenShift E2E test will be added that will verify that the `ScyllaDBMonitoring` CRD can be used to integrate
+  with a Prometheus instance managed by OpenShift's user-workload monitoring stack (with bearer token).
+
+### Upgrade / Downgrade Strategy
+
+No specific upgrade or downgrade strategy is required.
+
+### Version Skew Strategy
+
+To create `ServiceMonitor` and `PrometheusRule` resources, Operator uses
+`github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring` Go module for their schema. In an OpenShift
+cluster, mentioned CRDs are already installed (along with the Prometheus Operator) therefore we can only support
+OpenShift clusters with the Prometheus Operator supporting the version of the `ServiceMonitor` and `PrometheusRule`
+resources we use in the Operator. Their API is stable enough to not expect any breaking changes in the foreseeable
+future, but it's worth noting that in our documentation (i.e., in
+the [Support matrix](https://operator.docs.scylladb.com/stable/support/releases.html#support-matrix)).
+
+### Documentation plan
+
+- Add a guide that will walk users through the process of configuring the `ScyllaDBMonitoring` CRD to disable the
+  managed Prometheus instance and configure the managed Grafana instance to use an external Prometheus instance as a
```

Comment:
> the process of configuring the `ScyllaDBMonitoring` CRD to disable the managed Prometheus instance

This wording is doing some heavy-lifting. In order to _disable_ the managed Prom instance it would be building on top of something that has it _enabled_ in the first place. Even if that's how we see it (that we're introducing a knob to disable parts of our functionality), I don't think that such a mental model will work for the user.

I think that we need to think the other way around: reach a state in which we have a "set up monitoring" guide that has a fork on the road: "spin up Prometheus" and "connect to an external Prometheus".

Reply:
Yeah, I like the idea of a fork for bringing your own `ExeternallyManaged` Prometheus in a single guide explaining the `ScyllaDBMonitoring` CRD. We can discuss the exact structure under the actual documentation PR. I've changed the wording accordingly.

I also added an optional blog post idea that would describe using the externally managed Prometheus on OpenShift, to avoid the risk of our docs becoming outdated on OpenShift changes.